pkg/validating-webhooks/hiveextension/v1beta1/agentclusterinstall_admission_hook.go

```diff
@@ -0,0 +1,292 @@
+package v1beta1
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	hiveext "github.com/openshift/assisted-service/api/hiveextension/v1beta1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	log "github.com/sirupsen/logrus"
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	agentClusterInstallGroup    = "extensions.hive.openshift.io"
+	agentClusterInstallVersion  = "v1beta1"
+	agentClusterInstallResource = "agentclusterinstalls"
+
+	agentClusterInstallAdmissionGroup   = "admission.agentinstall.openshift.io"
+	agentClusterInstallAdmissionVersion = "v1"
+)
+
+var (
+	mutableFields = []string{"ClusterMetadata"}
```

Comment:
Eventually this should include things in `Compute`, for example so workers can be created with different settings for hyperthreading, or any other settings we put in there (like the custom ignition endpoint we discussed earlier today). I'm just stating this to confirm the design will accommodate that.

Reply:
What fields do you think can be changed after the installation started?
If you refer for day2 operations, then the `installAlreadyStarted` function will return false on Day2 clusters.

Reply:
Ah, so this is only preventing change during installation, but allows changes again after installation finishes? That'll work. It would help to make that distinction clear in the docs, such as `docs/hive-integration/README.md`.

Reply:
Done